### PR TITLE
Fix drag/drop support

### PR DESCRIPTION
**Pull request by BretJohnson**
Fix so that drag/drop support works properly with the Cocoa backend.
This PR includes these fixes:

1. DraggingUpdate passed the wrong coordinates. Now do the proper conversion from view coordinates to window coordinates, as expected by the Xwt client.
2. Get the INSDraggingInfo object in the proper way, with `Runtime.GetINativeObject<INSDraggingInfo> (dragInfo, owns: false)`, same as we do in other code now.
3. Use the newer NSView.BeginDraggingSession API instead of the deprecated NSView.DragImage. This is needed so that we can provide a DraggingSource object in order to be notified when the drag/drop is over (no matter if successful or not) and call the Finished callback, as expected by Xwt client code.
4. Populate the pasteboard item used for the drag properly, supporting text drag/drop and internal formats (normally a serialized object where the Id is the assembly qualified type name).

To my knowledge the Android Designer is the only client that still uses this drag/drop code. All of these fixes were to make Android Designer drag/drop work properly.

### CODE REVIEW

File: Xwt.XamMac/Xwt.Mac/ViewBackend.cs [hunk 1]
```diff
@@ -617,10 +617,10 @@ static NSDragOperation DraggingUpdated (IntPtr sender, IntPtr sel, IntPtr dragIn
 			if (ob == null)
 				return NSDragOperation.None;
 			var backend = ob.Backend;
-			
-			INSDraggingInfo di = (INSDraggingInfo) Runtime.GetNSObject (dragInfo);
+
+			var di = Runtime.GetINativeObject<INSDraggingInfo> (dragInfo, owns: false);
```
Comment (Therzok) [inline Xwt.XamMac/Xwt.Mac/ViewBackend.cs:621]: These changes LGTM!

**Review by Therzok — APPROVED** [review]

File: Xwt.XamMac/Xwt.Mac/ViewBackend.cs [hunk 2]
```diff
@@ -883,5 +909,60 @@ public void UpdateChildPlacement ()
 			child.Frame = new CGRect ((nfloat)cx, (nfloat)cy, (nfloat)cwidth, (nfloat)cheight);
 		}
 	}
+
+	class DraggingSource : NSDraggingSource
```
Comment (Therzok) [inline Xwt.XamMac/Xwt.Mac/ViewBackend.cs:913]:
```suggestion
	class DraggingSource : INSDraggingSource
```

Comment (BretJohnson) [reply Xwt.XamMac/Xwt.Mac/ViewBackend.cs]: Fixed (now it's `class DraggingSource : NSObject, INSDraggingSource`.  I believe that's correct (and it seems to work) but feel free to confirm.

**Comment by BretJohnson** [comment]
Removing draft, as this should finally be ready to merge. @netonjm , @sandyarmstrong  - can you please also review.

**Review by Therzok — APPROVED** [review]

**Review by sandyarmstrong — APPROVED** [review]
Great work, thanks for this!

**Comment by sandyarmstrong** [comment]
I'll let @netonjm coordinate merging/bumping.

**Comment by BretJohnson** [comment]
@sevoku - This should be ready to merge, as soon as you approve.
Once merged, I'll include it in the Xwt bump here: https://github.com/xamarin/vsmac/pull/6485.

**Comment by BretJohnson** [comment]
@netonjm - Did you want @sevoku to look at anything in particular here? This has gotten a few other reviews already & I believe we're good.
In any case, it would great to get this merged as soon as we can, as not having it is something of a blocker for us. Thanks.

**Comment by netonjm** [comment]
> @netonjm - Did you want @sevoku to look at anything in particular here? This has gotten a few other reviews already & I believe we're good.
> In any case, it would great to get this merged as soon as we can, as not having it is something of a blocker for us. Thanks.

I added to add a Final review, but from my side is ready to go

**Pull request merged**